.vitepress/theme/components/PlaygroundBadge.vue

```diff
@@ -0,0 +1,190 @@
+<script setup lang="ts">
+interface Props {
+  name: string;
+  kind: 'correct' | 'incorrect';
+  rules?: Record<string, string | number | [string, string | number]> | undefined;
+  files?: Array<string> | undefined;
+  packages?: Record<string, string> | undefined;
+}
+// @ts-ignore
+withDefaults(defineProps<Props>(), {})
+
+import { zlibSync, strToU8, strFromU8 } from 'fflate';
+// @ts-ignore
+import { data } from '../../../tools/lint/examples/examples.data.ts';
+
+const configFileName = ".eslintrc.json";
+const packageJsonFileName = "package.json";
+
+const defaultConfig: any = {
+    "extends": ["plugin:@sap/cds/recommended"],
+    "rules": {}
+}
+
+const defaultPackageJson = {
+    "name": "My CAP project",
+    "version": "1.0.0",
+    "dependencies": {
+        "@sap/cds": "^7.0.0",
+    },
+    "devDependencies": {
+        "eslint": ">=7.0.0",
+        "@sap/eslint-plugin-cds": "^2.6.7", // Get version
+    }
+}
+
+function mergeJSONs(target: any, add: any) {
+    const isObject = (obj: unknown) => typeof obj === 'object';
+    Object.entries(add).forEach(([key, addVal]) => {
+        const targetVal = target[key];
+        if (targetVal && isObject(targetVal) && isObject(addVal)) {
+            if ((Array.isArray(targetVal) && Array.isArray(addVal))) {
+                targetVal.push(...addVal);
+                return;
+            }
+            mergeJSONs(targetVal, addVal);
+        } else {
+            target[key] = addVal;
+        }
+    });
+    return target;
+}
+
+function link(name: Props['name'] = "", kind: Props['kind'], rules?: Props['rules'], files?: Props['files'], packages?: Props['packages'] ): string {
+  let json = {};
+  const sources = {} as Record<string, string>;
+  if (rules) {
+    for (const [key, value] of Object.entries(rules)) {
+      defaultConfig.rules[key] = value;
+    }
+  }
+  sources[configFileName] = prettyStringify(defaultConfig);
+  if (packages) {
+    json = mergeJSONs(defaultPackageJson, packages);
+  } else {
+    json = defaultPackageJson;
+  }
+  sources[packageJsonFileName] = prettyStringify(json);
+  if (files) {
+    for (const file of files || []) {
```

Comment:
```suggestion
    for (const file of files) {
```
you already check for truthiness in line 68.